apps/src/StudioApp.js

```diff
@@ -811,8 +813,15 @@ StudioApp.prototype.handleClearPuzzle = function (config) {
     annotationList.clearRuntimeAnnotations();
   }
   if (config.afterClearPuzzle) {
-    config.afterClearPuzzle();
+    promise = config.afterClearPuzzle(config);
+  }
+  if (!promise) {
```

Comment:
I feel like it starts heading into spaghetti-code territory when afterClearPuzzle might return void or might return a promise. It looks like there are 5 implementations of afterClearPuzzle (applab.js, GameLab.js, studio.js, blockly/js/gamelab.js, studio.js) and my preference would be to add a one liner in each to return a promise. Could have a wrapper in the base like CreateInstaPromise() to make it easy to read.